### PR TITLE
Close issues and PR for inactivity after a certain amount of time.

### PR DESCRIPTION
**Pull request by khanhntd**
**Description:** <Describe what has changed. 
Relating to issues: 
Since we don't want issues to keep-going on-and-on without the cx reply back. Therefore, it would be good for us to close the issues and PR after a certain amount of time

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/aws-observability/aws-otel-collector/issues/713

**Testing:** < Describe what testing was performed and which tests were added.>
Test on my local branch and need to wait for 1 day to pass 
**Documentation:** < Describe the documentation added.>
https://github.com/actions/stale

### CODE REVIEW

File: .github/workflows/soaking.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 name: 'weekly soaking test'
 on:
   schedule:
-    - cron: '0 10 * * SAT' # every Saturday at 10 am UTC: pst 2am
+    - cron: '0 10 * * SAT' # every Saturday at 10 am UTC: pst 2am'0 10 * * SAT'
```
Comment (sethAmazon) [inline .github/workflows/soaking.yml:17]: Why add the cron in the comment?

File: .github/workflows/stale-bot.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          #Github stale actions: https://github.com/actions/stale
+          stale-issue-message: 'This issue is stale because it has been open 25 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+          stale-pr-message: 'This PR is stale because it has been open 30 days with no activity.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          days-before-issue-stale: 25 #Mark the issues as after 25 days
+          days-before-pr-stale: 30 #Mark the PR as stale after 30 days
+          days-before-issue-close: 5 #Close the issue if the issue has been marked as stall for 5 days
+          days-before-pr-close: -1 #Never close down the PR and keep the label stall on the PR
+          debug-only: true #Debugging purpose. Moreover, need to set secret ACTIONS_STEP_DEBUG = true to enable this
```
Comment (sethAmazon) [inline .github/workflows/stale-bot.yml:38]: Did you set up the secret in the main repo?

File: .github/workflows/soaking.yml (hunk 1 above)
Comment (khanhntd) [reply .github/workflows/soaking.yml]: Another typo from me. Should be deleted by now.

File: .github/workflows/stale-bot.yml (hunk 2 above)
Comment (khanhntd) [reply .github/workflows/stale-bot.yml:38]: Currently, I don't have permission to set up the secrets but will raise this to @alolita.

Comment (khanhntd) [reply .github/workflows/stale-bot.yml:38]: Not yet. I have raised to @alolita and waited for her to respond back

File: .github/workflows/soaking.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@
 name: 'weekly soaking test'
 on:
   schedule:
-    - cron: '0 10 * * SAT' # every Saturday at 10 am UTC: pst 2am
+    - cron: '0 10 * * SAT' # every Saturday at 10 am UTC: pst 2am'
```
Comment (SaxyPandaBear) [inline .github/workflows/soaking.yml:17]: Looks like this added a typo?

File: .github/workflows/stale-bot.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+name: 'Close stale issues and PR '
+on:
+  schedule:
+    - cron: '0 20 * * SUN' # every Sunday at 20 am UTC: PST 0:00 AM "
+    #- cron: '* * * * *' # For every minute and only for testing purpose"
```
Comment (SaxyPandaBear) [inline .github/workflows/stale-bot.yml:18]: We can delete this. No point in having it here for testing purposes once it gets merged.

**Comment by jefchien** [comment]
Have you taken a look at other stale bots? Like in [cloudwatch-agent](https://github.com/aws/amazon-cloudwatch-agent/blob/master/.github/workflows/close-stalled.yml) or [aws-sdk-go-v2](https://github.com/aws/aws-sdk-go-v2/blob/main/.github/workflows/stale_issue.yml) or [opentelemetry collector](https://github.com/open-telemetry/opentelemetry-collector/blob/main/.github/workflows/stale-pr.yaml).

File: .github/workflows/soaking.yml (hunk 3 above)
Comment (khanhntd) [reply .github/workflows/soaking.yml]: Another typo from me and fixed it. Thanks for the notice.

File: .github/workflows/stale-bot.yml (hunk 4 above)
Comment (khanhntd) [reply .github/workflows/stale-bot.yml]: Yes. I intend to delete it after discussing in the standup to know more about our standard of marking staling and closing.

**Comment by khanhntd** [comment]
> Have you taken a look at other stale bots? Like in [cloudwatch-agent](https://github.com/aws/amazon-cloudwatch-agent/blob/master/.github/workflows/close-stalled.yml) or [aws-sdk-go-v2](https://github.com/aws/aws-sdk-go-v2/blob/main/.github/workflows/stale_issue.yml) or [opentelemetry collector](https://github.com/open-telemetry/opentelemetry-collector/blob/main/.github/workflows/stale-pr.yaml).

Yes. Looked at it but intend to raise in the standup today to know more about our standard.

File: .github/workflows/stale-bot.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          #Github stale actions: https://github.com/actions/stale
+          #Message
+          stale-issue-message: 'This issue is stale because it has been open 25 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+          stale-pr-message: 'This PR is stale because it has been open 30 days with no activity.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          #Labels
+          stale-issue-label: closing-soon #Mark the issue as closing-soon if staling for 25 days
+          stale-pr-label: no-pr-activity #Mark the pr as no-pr-activity if staling for 30 days
+          #Days required
+          days-before-issue-stale: 25 #Mark the issues as after 25 days
```
Comment (sethAmazon) [inline .github/workflows/stale-bot.yml:38]: From @alolita maybe set stale at 30 and close at 60

**Review by sethAmazon — APPROVED** [review]

File: .github/workflows/stale-bot.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+name: 'Close stale issues and PR '
```
Comment (jefchien) [inline .github/workflows/stale-bot.yml:14]:
```suggestion
name: 'Close stale issues and PR'
```

File: .github/workflows/stale-bot.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+          #Labels
+          stale-issue-label: stale #Mark the issue as closing-soon if staling for 60 days
+          stale-pr-label: stale #Mark the pr as no-pr-activity if staling for 30 days
+          #Days required
+          days-before-issue-stale: 60 #Mark the issues as after 60 days
+          days-before-pr-stale: 30 #Mark the PR as stale after 30 days
+          days-before-issue-close: 30 #Close the issue if the issue has been marked as stall for 30 days
+          days-before-pr-close: -1 #Never close down the PR and keep the label stall on the PR
+          #Optionals
+          #debug-only: true #Debugging purpose. No actions will be taken if this flag turn on.
```
Comment (jefchien) [inline .github/workflows/stale-bot.yml:43]: Remove unused properties.

File: .github/workflows/stale-bot.yml (hunk 6 above)
Comment (khanhntd) [reply .github/workflows/stale-bot.yml]: Done for the change.

File: .github/workflows/stale-bot.yml (hunk 7 above)
Comment (khanhntd) [reply .github/workflows/stale-bot.yml]: Done for the change too.

**Pull request merged**